native/src/components/Header.tsx

```diff
@@ -82,14 +88,33 @@ const Header = ({
   const [previousRoute] = useState(navigation.getState().routes[navigation.getState().routes.length - 2])
   const [canGoBack] = useState(navigation.canGoBack())
 
+  const getShareTitle = (): string => {
+    const pageTitle = (route.params as { title?: string } | undefined)?.title
+    const poisRouteParams = route.params as RoutesParamsType[PoisRouteType]
+    if (route.name === POIS_ROUTE && poisRouteParams.multipoi) {
+      return t('pois:multiPois')
+    }
+
+    if (route.name === NEWS_ROUTE) {
+      const newsType = (route.params as { newsType: TuNewsType | LocalNewsType }).newsType
+      if (newsType === TU_NEWS_TYPE) {
+        return tunewsLabel
+      }
+      return t('localNews')
+    }
+
+    return pageTitle ?? t(route.name)
+  }
```

Comment:
Sorry I did not realize that earlier but the share title should always just be the pageTitle/fallback to `route.name` controlled by the separate routes instead deciding here in the header what would be a good share title. That way:
- page title and share title are consistent
- it is controlled by the routes and therefore easier extendable
- No weird typecasts necessary

To fix the few occurrences (AFAICS only local news list, tünews list and tünews details are wrong), you can use `useSetRouteTitle` and set the correct route titles that way.

Reply:
hm but this `useSetRouteTitle` is not used on everyRoute f.e. Disclaimer etc. so i have to add it there too, i guess

Reply:
exactly, route title with a fallback to `t(route.name)`

Reply:
i think the main issue in general is with the News overview pages that if we make a `GoBack` we don't get the correct route title.
Shouldn't it just be possible to use the 
`useSetRouteTitle({ navigation, title: <title> })` in News Overview and Details pages and pass there the correct title?

Reply:
I think its overwritten by the usage of  `useSetRouteTitle` in `News` component